### DISCUSSION TITLE
break down button variants

### DISCUSSION THREAD

**Discussion by hariDasu** [NJIT-IS332-Fall2023/is322-next-app#17] [Ideas]
Primary

https://www.figma.com/file/oL4OC1GAgUpaSE0TPqlifl/MARVEL-APP-(Community)?type=design&node-id=0-340&mode=dev

Secondary

https://www.figma.com/file/oL4OC1GAgUpaSE0TPqlifl/MARVEL-APP-(Community)?type=design&node-id=0-341&mode=dev

LinkedText

OauthFB

OauthGoogle

**Comment by hariDasu** [comment]
https://planningpokeronline.com/KprtG3KovJWZvsGG1u8R/

**Comment by ksl29** [comment]
shadcn-ui uses a library called class-variance-authority to create different variants for elements in their components. We could modify their components to have these different variants or we could use the same package they use directly.

**Comment by hariDasu** [comment]
@kadeem-lewis you seem like you can take the lead on this, in both picking a UI lib that would best fit our use case and opening first PR for this ticket

**Comment by ksl29** [comment]
do I need to wait for the completion of #7 before implementing this?

**Comment by hariDasu** [comment]
> do I need to wait for the completion of #7 before implementing this?

no, we can update the commit/pr flow after i finish this today/this weekend

**Comment by hariDasu** [comment]
https://dev.to/remusris/t3-stack-template-supabase-w-auth-db-and-shadcn-ui-basic-setup-2bl9
